### ISSUE TITLE
Cannot create Azure VM with Ubuntu Jammy stemcell

### ISSUE THREAD

**Issue by MSSedusch** [cloudfoundry/bosh-linux-stemcell-builder#234]
While working on #231 I also tested this fix for Ubuntu Jammy but provisioning of a VM would always time out.
I also tested the latest Ubuntu Jammy stemcell from the official repository (https://bosh.cloudfoundry.org/stemcells/) with the same result.

The root cause seems to be that the network does not work after the first boot of the VM. I tested with Accelerated Networking enabled and disabled.

If I connect to the VM via the serial console and execute ifdown & ifup, the VM can connect to the internet and the Azure wireserver (to report health).

Has anyone tested the Ubuntu Jammy stemcell on Azure?

**Comment by ystros** [comment]
We've been seeing similar behavior. Tracing through the Window Azure Linux Agent source code yesterday, we noticed that it is calling `ifdown eth0 && ifup eth0` as part of its provisioning code. It ends up marking the VM as provisioned, but our theory is that the network never comes back up and it gets stuck trying to report back to the Azure API.

See:
* Provisioning method: https://github.com/Azure/WALinuxAgent/blob/v2.6.0.2/azurelinuxagent/pa/provision/default.py#L213
* `publish_hostname`: https://github.com/Azure/WALinuxAgent/blob/v2.6.0.2/azurelinuxagent/common/osutil/default.py#L1176-L1180
* `restart_if`: https://github.com/Azure/WALinuxAgent/blob/v2.6.0.2/azurelinuxagent/common/osutil/default.py#L1163-L1174

Relevant logs from the VM. You can see the network go down then supposedly come up. Also of note (may not be relevant), some of the messages from waagent come after it runs `ifup eth0`, but the corresponding `systemd-networkd[302]: eth0: Link UP` comes after the last waagent message. waagent never makes it to the "Provisioning complete" message, nor a "Provisioning failed" message.
```
Jul 14 18:48:37 bosh-stemcell python3[1686]: 2022-07-14T18:48:37.879806Z INFO Daemon Daemon Azure Linux Agent Version:2.6.0.2
Jul 14 18:48:37 bosh-stemcell python3[1686]: 2022-07-14T18:48:37.884777Z INFO Daemon Daemon OS: ubuntu 22.04
Jul 14 18:48:37 bosh-stemcell python3[1686]: 2022-07-14T18:48:37.888403Z INFO Daemon Daemon Python: 3.10.4
Jul 14 18:48:37 bosh-stemcell python3[1686]: 2022-07-14T18:48:37.892400Z INFO Daemon Daemon Run daemon
Jul 14 18:48:37 bosh-stemcell python3[1686]: 2022-07-14T18:48:37.896165Z INFO Daemon Daemon Using waagent for provisioning
Jul 14 18:48:37 bosh-stemcell python3[1686]: 2022-07-14T18:48:37.900856Z INFO Daemon Daemon Clean protocol and wireserver endp>
Jul 14 18:48:37 bosh-stemcell python3[1686]: 2022-07-14T18:48:37.905763Z INFO Daemon Daemon Running default provisioning handl>
Jul 14 18:48:37 bosh-stemcell python3[1686]: 2022-07-14T18:48:37.916036Z INFO Daemon Daemon Unable to get cloud-init enabled s>
Jul 14 18:48:37 bosh-stemcell python3[1686]: 2022-07-14T18:48:37.940056Z INFO Daemon Daemon Unable to get cloud-init enabled s>
Jul 14 18:48:37 bosh-stemcell python3[1686]: 2022-07-14T18:48:37.940352Z INFO Daemon Daemon cloud-init is enabled: False
Jul 14 18:48:37 bosh-stemcell python3[1686]: 2022-07-14T18:48:37.941140Z INFO Daemon Daemon Copying ovf-env.xml
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.054453Z INFO Daemon Daemon Successfully mounted dvd
Jul 14 18:48:38 bosh-stemcell systemd[1]: mnt-cdrom-secure.mount: Deactivated successfully.
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.103425Z INFO Daemon Daemon Detect protocol endpoint
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.104502Z INFO Daemon Daemon Clean protocol and wireserver endp>
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.105264Z INFO Daemon Daemon WireServer endpoint is not found. >
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.105931Z INFO Daemon Daemon Test for route to 168.63.129.16
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.106785Z INFO Daemon Daemon Route to 168.63.129.16 exists
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.107353Z INFO Daemon Daemon Wire server endpoint:168.63.129.16
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.130958Z INFO Daemon Daemon Fabric preferred wire protocol ver>
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.142575Z INFO Daemon Daemon Wire protocol version:2012-11-30
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.147253Z INFO Daemon Daemon Server preferred version:2015-04-05
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.495577Z INFO Daemon Daemon Initializing goal state during pro>
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.507967Z INFO Daemon Daemon Forcing an update of the goal stat>
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.512980Z INFO Daemon Daemon Fetching goal state [incarnation 1]
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.561423Z INFO Daemon Daemon Certificate with thumbprint 419EFA>
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.561917Z INFO Daemon Daemon Fetch goal state completed
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.609833Z INFO Daemon Daemon Fetched new vmSettings [correlatio>
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.610366Z INFO Daemon Daemon Status Blob type 'None' is not val>
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.628817Z INFO Daemon Daemon Starting provisioning
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.629084Z INFO Daemon Daemon Handle ovf-env.xml.
Jul 14 18:48:38 bosh-stemcell python3[1686]: 2022-07-14T18:48:38.630663Z INFO Daemon Daemon Set hostname [pkrvme9fivq18aj]
Jul 14 18:48:38 pkrvme9fivq18aj python3[1686]: 2022-07-14T18:48:38.640415Z INFO Daemon Daemon Publish hostname [pkrvme9fivq18a>
Jul 14 18:48:38 pkrvme9fivq18aj python3[1686]: 2022-07-14T18:48:38.646352Z INFO Daemon Daemon Examine /proc/net/route for prim>
Jul 14 18:48:38 pkrvme9fivq18aj python3[1686]: 2022-07-14T18:48:38.651557Z INFO Daemon Daemon Primary interface is [eth0]
Jul 14 18:48:38 pkrvme9fivq18aj systemd-networkd[302]: eth0: Link DOWN
Jul 14 18:48:38 pkrvme9fivq18aj dhclient[393]: receive_packet failed on eth0: Network is down
Jul 14 18:48:38 pkrvme9fivq18aj systemd-networkd[302]: eth0: Lost carrier
Jul 14 18:48:38 pkrvme9fivq18aj python3[1686]: 2022-07-14T18:48:38.780704Z INFO Daemon Daemon Create user account if not exists
Jul 14 18:48:38 pkrvme9fivq18aj python3[1686]: 2022-07-14T18:48:38.781096Z INFO Daemon Daemon User vcap already exists, skip u>
Jul 14 18:48:38 pkrvme9fivq18aj python3[1686]: 2022-07-14T18:48:38.781841Z INFO Daemon Daemon Configure sudoer
Jul 14 18:48:38 pkrvme9fivq18aj python3[1686]: 2022-07-14T18:48:38.787293Z INFO Daemon Daemon Configure sshd
Jul 14 18:48:38 pkrvme9fivq18aj python3[1686]: 2022-07-14T18:48:38.787817Z INFO Daemon Daemon Disabled SSH password-based auth>
Jul 14 18:48:38 pkrvme9fivq18aj python3[1686]: 2022-07-14T18:48:38.788922Z INFO Daemon Daemon Configured SSH client probing to>
Jul 14 18:48:38 pkrvme9fivq18aj python3[1686]: 2022-07-14T18:48:38.790326Z INFO Daemon Daemon Deploy ssh public key.
Jul 14 18:48:38 pkrvme9fivq18aj systemd-networkd[302]: eth0: Link UP
Jul 14 18:48:38 pkrvme9fivq18aj systemd-networkd[302]: eth0: Gained carrier
Jul 14 18:48:38 pkrvme9fivq18aj systemd[1]: Starting Update resolvconf for networkd DNS...
Jul 14 18:48:38 pkrvme9fivq18aj sh[1762]: sed: can't read /run/systemd/netif/leases/*: No such file or directory
Jul 14 18:48:39 pkrvme9fivq18aj sh[1794]: sed: can't read /run/systemd/netif/leases/*: No such file or directory
```

It should also be noted that from Bionic to Jammy stemcells, the version of waagent was upgraded from v2.2.53 to v2.6.0.2

**Comment by MSSedusch** [comment]
seems to work now with #235. Thanks @nouseforaname

**Comment by ramonskie** [comment]
just released jammy 0.3 which includes this fix
